openhands/agenthub/codeact_agent/function_calling.py

```diff
@@ -263,9 +263,44 @@ def response_to_actions(
                         f'Missing required argument "task_list" for "plan" command in tool call {tool_call.function.name}'
                     )
 
+                raw_task_list = arguments.get('task_list', [])
+                if not isinstance(raw_task_list, list):
+                    raise FunctionCallValidationError(
+                        f'Invalid format for "task_list". Expected a list but got {type(raw_task_list)}.'
+                    )
+
+                # Normalize task_list to ensure it's always a list of dictionaries
+                normalized_task_list = []
+                for i, task in enumerate(raw_task_list):
+                    if isinstance(task, dict):
+                        # Task is already in correct format, ensure required fields exist
+                        normalized_task = {
+                            'id': task.get('id', f'task-{i + 1}'),
+                            'title': task.get('title', 'Untitled task'),
+                            'status': task.get('status', 'todo'),
+                            'notes': task.get('notes', ''),
+                        }
+                    elif isinstance(task, str):
```

Comment:
I think we can probably just raise fn call validation error here, no need to try to recover?